src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -132,6 +133,10 @@ Copyright (c) .NET Foundation. All rights reserved.
       </_ManagedAssembliesToLink>
     </ItemGroup>
 
+    <ItemGroup>
+      <_TrimmerFeatureSettings Include="@(RuntimeHostConfigurationOption)" Condition="'%(RuntimeHostConfigurationOption.Link)' != 'false'" />
```

Comment:
Do you want this to be opt-in or opt-out? From reading the spec, it sounds like it should be opt-in:

[https://github.com/dotnet/designs/blob/master/accepted/2020/feature-switch.md#generate-the-right-input-for-the-linker-in-sdk](https://github.com/dotnet/designs/blob/master/accepted/2020/feature-switch.md#generate-the-right-input-for-the-linker-in-sdk)

> All names/values from RuntimeHostConfigurationOption with Link == true metadata will be passed to the ILLink task

If you keep it being opt-out, I think we may need to modify some of the existing `RuntimeHostConfigurationOption` items. For example 

[https://github.com/dotnet/sdk/blob/b910664dfd9f4c723606e8caf18b4a69d16d2c71/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L373-L379](https://github.com/dotnet/sdk/blob/b910664dfd9f4c723606e8caf18b4a69d16d2c71/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L373-L379)

Those 2 options use non-boolean based values, which I believe will cause the linker to complain. The ILLink Task will [blindly copy any value to the command line](([https://github.com/mono/linker/blob/ba6d3da58065e0487eeb05c1a3d688afd43b94dc/src/ILLink.Tasks/LinkTask.cs#L352-L359](https://github.com/mono/linker/blob/ba6d3da58065e0487eeb05c1a3d688afd43b94dc/src/ILLink.Tasks/LinkTask.cs#L352-L359)). Then the linker.exe will [spit out an `Invalid argument` comment](https://github.com/mono/linker/blob/ba6d3da58065e0487eeb05c1a3d688afd43b94dc/src/linker/Linker/Driver.cs#L897) saying it can't parse it to a bool.

Reply:
I am going for opt-out. From the section of the spec where it introduces the metadata:

> RuntimeHostConfigurationOption items will have optional boolean Link metadata which indicates that the feature setting should be optimized by the linker. This metadata can be set to false either in the SDK (for features which don't have defined linker optimizations) or by developers (if they want to disable linking of a particular feature). The default behavior will be for boolean RuntimeHostConfigurationOptions to be treated as if Link == true, using ItemDefinitionGroup or an appropriate MSBuild mechanism.

You're right that we need to filter out non-booleans. I will do that and add a test for this case, and also update that misleading line in the spec.

Thanks!

Reply:
Note - here's another one that has a non-boolean value:

https://github.com/dotnet/docs/blob/master/docs/standard/globalization-localization/globalization-icu.md#app-local-icu

```xml
<ItemGroup>
  <RuntimeHostConfigurationOption Include="System.Globalization.AppLocalIcu" Value="<suffix>:<version> or <version>" />
</ItemGroup>
```

Reply:
Switched to opt-in here and in the spec. Thanks!